ADR/0033-product-version-controller.md

```diff
@@ -0,0 +1,493 @@
+      annotations:
+        pvc.konflux.dev/cloned-from: cool-comp1-main
+    spec:
+      application: "cool-app-{version}"
+      componentName: "cool-comp1-{version}"
+      source:
+        git:
+          context: {cool-comp1-context}
+          dockerfileUrl: {cool-comp1-dockerfileUrl}
+          revision: {cool-comp1-revision}
+          uri: git@github.com:example/comp1.git
+
+  - apiVersion: appstudio.redhat.com/v1alpha1
+    kind: Component
+    metadata:
+      name: "cool-comp2-{version}"
+      annotations:
+        pvc.konflux.dev/cloned-from: cool-comp2-main
+    spec:
+      application: "cool-app-{version}"
+      componentName: "cool-comp2-{version}"
+      source:
+        git:
+          context: {cool-comp2-context}
+          dockerfileUrl: {cool-comp2-dockerfileUrl}
+          revision: {cool-comp2-revision}
+          uri: git@github.com:example/comp2.git
+```
+
+Following is a new *ProjectDevelopmentStream* resource making use of the
+template above:
+
+```
+apiVersion: pvc.konflux.dev/v1alpha1
+kind: ProjectDevelopmentStream
+metadata:
+  name: my-cool-project-main
+  ownerReference:
+    apiVersion: pvc.konflux.dev/v1alpha1
+    kind: Project
+    name: my-cool-project
+spec:
+  project: my-cool-project
+  template:
+    name: my-cool-project-stream-template
+    values:
+    - name: version
+      value: v1.0.0
+    - name: cool-comp1-dockerfileUrl
+      value: Dockerfile-rhel10
+```
+
+Sine the *ProjectDevelopmentStream* resource is referencing a template, here are
+the resources that would get created as a result:
+
+```
+---
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: Application
+metadata:
+  name: "cool-app-v1.0.0"
+  ownerReference:
+    apiVersion: pvc.konflux.dev/v1alpha1
+    kind: ProjectDevelopmentStream
+    name: my-cool-project-v1.0.0
+  annotations:
+    pvc.konflux.dev/cloned-from: cool-app1-main
+spec:
+  displayName: "Cool App v1.0.0"
+---
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: Component
+metadata:
+  name: "cool-comp1-v1.0.0"
+  ownerReference:
+    apiVersion: appstudio.redhat.com/v1alpha1
+    kind: Application
+    name: "cool-app-v1.0.0"
+  annotations:
+    pvc.konflux.dev/cloned-from: cool-comp1-main
+spec:
+  application: "cool-app-v1.0.0"
+  componentName: "cool-comp1-v1.0.0"
+  source:
+    git:
+      context: ./
+      dockerfileUrl: Dockerfile-rhel10
+      revision: v1.0.0
+      uri: git@github.com:example/comp1.git
+---
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: Component
+metadata:
+  name: "cool-comp2-v1.0.0"
+  ownerReference:
+    apiVersion: appstudio.redhat.com/v1alpha1
+    kind: Application
+    name: "cool-app-v1.0.0"
+  annotations:
+    pvc.konflux.dev/cloned-from: cool-comp2-main
+spec:
+  application: "cool-app-v1.0.0"
+  componentName: "cool-comp2-v1.0.0"
+  source:
+    git:
+      context: ./
+      dockerfileUrl: Dockerfile
+      revision: fixed-rev
+      uri: git@github.com:example/comp2.git
+```
+
+## Consequences
+
+TBD.
```

Comment:
It would be good to fill this out.

* What is potentially more difficult if we adopt this decision?
* What is potentially easier?
* Which teams/subsystems are affected and need to take action?

Reply:
Added content to the Consequences section.